drivers/wireless/lpwan/rn2xx3/rn2xx3.c

```diff
@@ -0,0 +1,1564 @@
+  if (length < 0)
+    {
+      return length;
+    }
+
+  err = get_command_err(priv);
+  if (err < 0)
+    {
+      return err;
+    }
+
+  priv->mod = mod;
+  return 0;
+}
+
+/****************************************************************************
+ * Name: rn2xx3_getmod
+ *
+ * Description:
+ *    Get the modulation type of the RN2xx3.
+ *
+ ****************************************************************************/
+
+static int rn2xx3_getmod(FAR struct rn2xx3_dev_s *priv, FAR char *mod)
+{
+  ssize_t length;
+  char response[12];
+
+  length =
+      rn2xx3_getparam(priv, "radio get mod\r\n", response, sizeof(response));
+  if (length < 0)
+    {
+      return length;
+    }
+
+  /* Copy over modulation */
+
+  memcpy(mod, response, length);
+  return 0;
+}
+
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+
+/****************************************************************************
+ * Name: rn2xx3_open
+ ****************************************************************************/
+
+static int rn2xx3_open(FAR struct file *filep)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct rn2xx3_dev_s *priv = inode->i_private;
+  int err;
+
+  err = nxmutex_lock(&priv->devlock);
+  if (err < 0)
+    {
+      return err;
+    }
+
+  /* Only one user at a time */
+
+  if (priv->crefs > 0)
+    {
+      err = -EBUSY;
+      wlerr("Too many fds");
+      goto early_ret;
+    }
+
+  priv->crefs++;
+  wlinfo("crefs: %d\n", priv->crefs);
+
+early_ret:
+  nxmutex_unlock(&priv->devlock);
+  return err;
+}
+
+#endif
+
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+
+/****************************************************************************
+ * Name: rn2xx3_close
+ ****************************************************************************/
+
+static int rn2xx3_close(FAR struct file *filep)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct rn2xx3_dev_s *priv = inode->i_private;
+  int err;
+
+  err = nxmutex_lock(&priv->devlock);
+  if (err < 0)
+    {
+      return err;
+    }
+
+  priv->crefs--;
+  wlinfo("crefs: %d\n", priv->crefs);
+  nxmutex_unlock(&priv->devlock);
+  return err;
+}
+
+#endif
+
+/****************************************************************************
+ * Name: rn2xx3_read
+ *
+ * Description:
+ *   Puts the RN2xx3 into continuous receive mode and waits for data to be
+ *   received before returning it.
+ *
+ ****************************************************************************/
+
+static ssize_t rn2xx3_read(FAR struct file *filep, FAR char *buffer,
+                           size_t buflen)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct rn2xx3_dev_s *priv = inode->i_private;
+  ssize_t length = 0;
+  ssize_t received = 0;
+  static const char receive_cmd[] = "radio rx 0\r\n";
+  char response[30];
+  char hex_byte[3] =
+  {
+    0, 0 /* Extra space for null terminator */
+  };
+
+  /* Exclusive access */
+
+  length = nxmutex_lock(&priv->devlock);
```

Comment:
I am not really a fun of variable reuse like that. Why not just declare `int ret` and use that for generic functions? Optimizes will sort this all out anyway.

Reply:
Reason being that ssize_t can encompass the values of `int ret`. Having one variable containing these errors allows me to `goto early_ret` and always unlock the mutex and return an error when one is present, without selecting between whether the error is in `int ret` or `ssize_t length`. I think this reuse is fine considering all the `*read` and `*write` function indicate an error in this same way, through a `ssize_t`.

Reply:
I meant you should have 2-3 variables. `int ret`, `ssize_t nwritten`, `ssize_t nread` and use them accordingly. Seeing `length = nxmutex` makes me question what I see. Why is mutex returning length? You know what I mean? If you had just one variable `ssize_t ret` that would be less confusing.

Reply:
You must always prioritize READING the code, not ease of writing.